June release notes (#110)
ShiftLeftSecurity/documentation: release-notes/june-2019.md

# ShiftLeft June 2019 Updates

## ShiftLeft Ocular

> * **On-Disk Code Property Graph (CPG) Support**. Disk overflow is automatically enabled for CPGs. Disk overflow moves cache entries to disk only when the number of entries in memory exceed the maximum memory allocation. Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other sections of the CPG are swapped onto disk. The following (example) information is displayed by ShiftLeft Ocular when disk overflow is enabled

Reviewer: ```suggestion
* **On-Disk Code Property Graph (CPG) Support**. Disk overflow is automatically enabled for CPGs. Disk overflow moves parts of the CPG to disk only when the number of entries in memory reach 80% of the memory allocation. Based on the query, ShiftLeft Ocular automatically identifies the parts of the CPG needed, which are loaded into RAM. The following (example) information is displayed by ShiftLeft Ocular when disk overflow is enabled
```

Author: Done

> * **On-Disk Code Property Graph (CPG) Support**. Disk overflow is automatically enabled for CPGs. Disk overflow moves cache entries to disk only when the number of entries in memory exceed the maximum memory allocation. Based on the query, ShiftLeft Ocular identifies the parts of the CPG needed, which are loaded into RAM. All other sections of the CPG are swapped onto disk. The following (example) information is displayed by ShiftLeft Ocular when disk overflow is enabled

Reviewer: We should add which version of Ocular introduced the support for this.

Author: Not sure which version did. Do you know?

Reviewer: Ask Michael

Author: Done

     ```
     INFO heap usage (after GC) was 0.87 -> scheduled to clear 100000 references (asynchronously)
     INFO attempting to clear 100000 references
     ```

>      The result is that larger CPGs are now supported, though at a performance cost. For large applications, a better approach is to optimize performance through the appropriate combination of memory and disks sizes.

Reviewer: ```suggestion
     The result is that larger CPGs are now supported, though at a performance cost.
```

Author: Done

> * **Recommendations on Memory Size**. ShiftLeft Ocular now displays information on the amount of memory needed to create or load a CPG, allowing you to [optimize performance for large applications](../using-ocular/ocular-memory-size.md), instead of defaulting to disk overflow.

Reviewer: We should include a sample output of such a recommendation.

Author: How can I get a sample output?

Reviewer: Michael can provide one

Author: Done

> * **Recommendations on Memory Size**. ShiftLeft Ocular now displays information on the amount of memory needed to create or load a CPG, allowing you to [optimize performance for large applications](../using-ocular/ocular-memory-size.md), instead of defaulting to disk overflow.

Reviewer: We should include the Ocular version that introduced this feature.

Author: Not sure which version did. Do you know?

Reviewer: Please ask Michael

Author: Done

* **Common Queries**. The ShiftLeft Ocular documentation has been improved with information on the most common queries:

   - [Analyzing Dependencies](../using-ocular/common-queries/dependency-analysis.md)
   - [Analyzing Types and Packages](../using-ocular/common-queries/types-packages-analysis.md)
   - [Analyzing Method Parameters](../using-ocular/common-queries/parameters-analyze.md)
   - [Analyzing Methods](../using-ocular/common-queries/methods-analyze.md)
   - [Uncovering Data Flows](../using-ocular/common-queries/data-flows.md)

## ShiftLeft Protect 

* **ShiftLeft Protect Event Viewer**. From the [Vulnerabilities Dashboard](../using-inspect-protect/using-workflow/vulnerability-dashboard.md#event-details), you can now view details on a sampling of security events for a vulnerability. This includes information on an event's headers, payloads, request origin, and whether the event was blocked by ShiftLeft Protect.  Such information can help you in remediation, to make precise code fixes and to determine the extent of an attack. 

>      Note that by default the Event Viewer only displays an event count summary. In order to include the full payloads of attack events, you must turn on the CLI Flag [sec.collect.attack.info](../using-inspect-protect/protect-java/configuring-the-microagent.md#collect-attack-information). User opt-in is required because collected event information collected potentially sensitive, and consumes more network resources to transmit that data to the ShiftLeft infrastructure.

Reviewer: "the CLI Flag" is actually an agent configuration item, not really a CLI flag.

Author: Done

# ShiftLeft June 2019 Updates

## ShiftLeft Ocular

* **On-Disk Code Property Graph (CPG) Support**. (Introduced in ShiftLeft Ocular version 0.3.38) Disk overflow is automatically enabled for CPGs. Disk overflow moves parts of the CPG to disk only when the number of entries in memory reach 80% of the maximum memory allocation. Based on the query, ShiftLeft Ocular automatically identifies the parts of the CPG needed, which are loaded into RAM. The following (example) information is displayed by ShiftLeft Ocular when disk overflow is enabled

     ```
     INFO heap usage (after GC) was 0.87 -> scheduled to clear 100000 references (asynchronously)
     INFO attempting to clear 100000 references
     ```
     The result is that larger CPGs are now supported, though at a performance cost. 
 

> * **Recommendations on Memory Size**. (Introduced in ShiftLeft Ocular version 0.3.41) ShiftLeft Ocular now displays information on the amount of memory needed to create or load a CPG, allowing you to [optimize performance for large applications](../using-ocular/ocular-memory-size.md), instead of defaulting to disk overflow. For example

Reviewer: ```suggestion
* **Recommendations on Memory Size**. (Introduced in ShiftLeft Ocular version 0.3.41) ShiftLeft Ocular now displays information on the amount of memory needed to create or load a CPG, allowing you to [optimize performance for large applications](../using-ocular/ocular-memory-size.md), instead of defaulting to disk overflow. Example output
```

Author: Done

     ```

>      The cpg.bin.zip you are loading is 63MB on disk and requires approximately 12712MB heap, 

Reviewer: We should add a sample command that the user may have issues to get this message, so that the reader has context on what this output is about.

Author: Done

Reviewer: I'm not seeing this

Author: I added this text "For example, using the command loadCpg to load a large CPG would display"

Reviewer: We should add what command the user issued before this text appears, to give context

Author: I asked Michael and he told me to use "loadcpg"

# ShiftLeft June 2019 Updates

## ShiftLeft Ocular

* **On-Disk Code Property Graph (CPG) Support**. (Introduced in ShiftLeft Ocular version 0.3.38) Disk overflow is automatically enabled for CPGs. Disk overflow moves parts of the CPG to disk only when the number of entries in memory reach 80% of the maximum memory allocation. Based on the query, ShiftLeft Ocular automatically identifies the parts of the CPG needed, which are loaded into RAM. The following (example) information is displayed by ShiftLeft Ocular when disk overflow is enabled

     ```
     INFO heap usage (after GC) was 0.87 -> scheduled to clear 100000 references (asynchronously)
     INFO attempting to clear 100000 references
     ```
     The result is that larger CPGs are now supported, though at a performance cost. 
 
* **Recommendations on Memory Size**. (Introduced in ShiftLeft Ocular version 0.3.41) ShiftLeft Ocular now displays information on the amount of memory needed to create or load a CPG, allowing you to [optimize performance for large applications](../using-ocular/ocular-memory-size.md), instead of defaulting to disk overflow. For example

     ```
     ocular> createCpg("someBigApp.jar")
 

>      ocular> The cpg.bin.zip you are loading is 63MB on disk and requires approximately 12712MB heap, 

Reviewer: ```suggestion
     The cpg.bin.zip you are loading is 63MB on disk and requires approximately 12712MB heap, 
```
